pkg/uidmap/uidmap.go

```diff
@@ -0,0 +1,136 @@
+package uidmap
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/docker/libcontainer/user"
+)
+
+func xlateOneFile(path string, finfo os.FileInfo, containerRoot uint32, inverse bool) error {
+	uid := uint32(finfo.Sys().(*syscall.Stat_t).Uid)
+	gid := uint32(finfo.Sys().(*syscall.Stat_t).Gid)
+	mode := finfo.Mode()
+
+	if (uid == containerRoot || gid == containerRoot) && !inverse {
```

Comment:
Check to make sure we aren't trying to translate an already translated image.  It could happen if a translated image is pushed to the registry or if there happens to be a UID in the image that conflicts with the UID we are using for docker-root.